waspc/data/Generator/templates/server/src/core/auth.js

```diff
@@ -28,68 +15,17 @@ const auth = handleRejection(async (req, res, next) => {
     return next()
   }
 
-  if (authHeader.startsWith('Bearer ')) {
-    const token = authHeader.substring(7, authHeader.length)
-    req.user = await getUserFromToken(token)
-  } else {
-    throwInvalidCredentialsError()
-  }
-
-  next()
-})
-
-export async function getUserFromToken(token) {
-  let userIdFromToken
-  try {
-    userIdFromToken = (await verify(token)).id
-  } catch (error) {
-    if (['TokenExpiredError', 'JsonWebTokenError', 'NotBeforeError'].includes(error.name)) {
-      throwInvalidCredentialsError()
-    } else {
-      throw error
-    }
-  }
+  const { session, user } = await getSessionAndUserFromBearerToken(req);
 
-  const user = await prisma.{= userEntityLower =}
-    .findUnique({
-      where: { id: userIdFromToken },
-      include: {
-        {= authFieldOnUserEntityName =}: {
-          include: {
-            {= identitiesFieldOnAuthEntityName =}: true
-          }
-        }
-      }
-    })
-  if (!user) {
+  if (!session || !user) {
```

Comment:
Curious, is it possible that one of these things is truthy and another is not?

Reply:
By the current `getSessionAndUserFromBearerToken` implementation it is not possible, but it felt weird only checking one of those things. 

Logically, it okay, both of those things need to be defined and that's why I'm checking it even if it's okay to check only one of the two.

Reply:
Ok, makes sense.